cmd/spire-agent/cli/run/run_test.go

```diff
@@ -78,26 +80,578 @@ func TestParseFlagsGood(t *testing.T) {
+			},
+		},
+		{
+			msg: "trust_domain should be correctly parsed",
+			input: func(c *config) {
+				c.Agent.TrustDomain = "foo"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.Equal(t, "spiffe://foo", c.TrustDomain.String())
+			},
+		},
+		{
+			msg:         "invalid trust_domain should return an error",
+			expectError: true,
+			input: func(c *config) {
+				c.Agent.TrustDomain = "i'm invalid"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.Nil(t, c)
+			},
+		},
+		{
+			msg: "socket_path should be correctly configured",
+			input: func(c *config) {
+				c.Agent.SocketPath = "foo"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.Equal(t, "foo", c.BindAddress.Name)
+				require.Equal(t, "unix", c.BindAddress.Net)
+			},
+		},
+		{
+			msg: "join_token should be correctly configured",
+			input: func(c *config) {
+				c.Agent.JoinToken = "foo"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.Equal(t, "foo", c.JoinToken)
+			},
+		},
+		{
+			msg: "data_dir should be correctly configured",
+			input: func(c *config) {
+				c.Agent.DataDir = "foo"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.Equal(t, "foo", c.DataDir)
+			},
+		},
+		{
+			msg: "enable_sds should be correctly configured",
+			input: func(c *config) {
+				c.Agent.EnableSDS = true
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.True(t, c.EnableSDS)
+			},
+		},
+		{
+			msg: "logger gets set correctly",
+			input: func(c *config) {
+				c.Agent.LogLevel = "WARN"
+				c.Agent.LogFormat = "TEXT"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.NotNil(t, c.Log)
+
+				l := c.Log.(*log.Logger)
+				require.Equal(t, logrus.WarnLevel, l.Level)
+				require.Equal(t, &logrus.TextFormatter{}, l.Formatter)
+			},
+		},
+		{
+			msg: "log_level and log_format are case insensitive",
+			input: func(c *config) {
+				c.Agent.LogLevel = "wArN"
+				c.Agent.LogFormat = "TeXt"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.NotNil(t, c.Log)
+
+				l := c.Log.(*log.Logger)
+				require.Equal(t, logrus.WarnLevel, l.Level)
+				require.Equal(t, &logrus.TextFormatter{}, l.Formatter)
+			},
+		},
+		{
+			msg:         "invalid log_level returns an error",
+			expectError: true,
+			input: func(c *config) {
+				c.Agent.LogLevel = "not-a-valid-level"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.Nil(t, c)
+			},
+		},
+		{
+			msg:         "invalid log_format returns an error",
+			expectError: true,
+			input: func(c *config) {
+				c.Agent.LogFormat = "not-a-valid-format"
+			},
+			test: func(t *testing.T, c *agent.Config) {
+				require.Nil(t, c)
+			},
+		},
+	}
+
+	for _, testCase := range cases {
+		input := defaultValidConfig()
+
+		testCase.input(input)
+
+		t.Run(testCase.msg, func(t *testing.T) {
+			ac, err := newAgentConfig(input)
+			if testCase.expectError {
+				require.Error(t, err)
```

Comment:
nit: in general i tend to prefer require.EqualError over require.Error as it asserts that its failing in the way I expect